File: Completo/ws16StoreApp/StoreApp_py_js_java/StoreApp.js/StoreApp/ec.edu.espe.store.model/Product.js
import { Tax } from '../utils/Tax.js';

export class Product {
    id;
    description;
    price;
    pvp;

    constructor(id, description, price, pvp = null) {
        this.id = id;
        this.description = description;
        this.price = price;
        
        if (pvp === null) {
            // //TODO COMPUTE TOTAL PRICE
            this.pvp = Tax.computeTotal(price, 15);
        } else {
            this.pvp = pvp;
        }
    }

    getId() { return this.id; }
    getDescription() { return this.description; }
    getPrice() { return this.price; }
    getPvp() { return this.pvp; }

    toString() {
        return `Product{id=${this.id}, description=${this.description}, price=${this.price}, pvp=${this.pvp}}`;
    }
}

File: Completo/ws16StoreApp/StoreApp_py_js_java/StoreApp.Java/StoreApp/StoreApp/src/ec/edu/espe/store/model/Product.java

package ec.edu.espe.store.model;

import utils.Tax;

/**
 *
 * @author LABS-ESPE
 */
public class Product {
    int id;
    String description;
    float price;

    public Product(int id, String description, float price) {
        this.id = id;
        this.description = description;
        this.price = price;
        //TODO COMPUTE TOTAL PRICE
        
       pvp=Tax.computeTotal(price, 15F);
        
    
        
        
    }

    
    public Product(int id, String description, float price, float pvp) {
        this.id = id;
        this.description = description;
        this.price = price;
        this.pvp = pvp;
    }

    
    
    
    @Override
    public String toString() {
        return "Product{" + "id=" + id + ", description=" + description + ", price=" + price + ", pvp=" + pvp + '}';
    }

    
    
    public int getId() {
        return id;
    }

    public String getDescription() {
        return description;
    }

    public float getPrice() {
        return price;
    }

    public float getPvp() {
        return pvp;
    }
    float pvp;
    
    
    
    
}
